src/Compilers/CSharp/Portable/Binder/Binder_Await.cs

```diff
@@ -286,7 +302,130 @@ private bool GetAwaitableExpressionInfo(
             TypeSymbol awaiterType = getAwaiter.Type!;
             return GetIsCompletedProperty(awaiterType, node, expression.Type!, diagnostics, out isCompleted)
                 && AwaiterImplementsINotifyCompletion(awaiterType, node, diagnostics)
-                && GetGetResultMethod(getAwaiter, node, expression.Type!, diagnostics, out getResult, out getAwaiterGetResultCall);
+                && GetGetResultMethod(getAwaiter, node, expression.Type!, diagnostics, out getResult, out getAwaiterGetResultCall)
+                && (!isRuntimeAsyncEnabled || getRuntimeAwaitAwaiter(awaiterType, out runtimeAsyncAwaitCall));
+
+            bool tryGetRuntimeAwaitHelper(out MethodSymbol? runtimeAwaitHelper)
```

Comment:
1. There are captures. This reduces readability and causes allocations. Consider reducing to only capture `this`. Same for `getRuntimeAwaitAwaiter` below
2. I'd suggest keeping the `isRuntimeAsyncEnabled` check at the call site (`if (isRuntimeAsyncEnabled && tryGetRuntimeAwaitHelper(...)`) for clarity